src/hooks/validation/models.ts

```diff
@@ -0,0 +1,102 @@
+import * as fs from 'fs';
+import * as path from 'path';
+
+export class SpecificationFile {
+  private readonly name: string;
+
+  constructor(name: string) {
+    const localDirectory = process.cwd();
+    this.name = path.join(localDirectory, name);
+  }
+
+  isNotValid(): boolean {
+    return !fs.existsSync(this.name) || !fs.lstatSync(this.name).isFile();
+  }
```

Comment:
I think that name `isNotFile` is better, because `isNotValid` can indicates also that spec isn't valid :)

```suggestion
  isNotFile(): boolean {
    return !fs.existsSync(this.name) || !fs.lstatSync(this.name).isFile();
  }
```

Reply:
Probably the name is not the best as you mention, but in the future this SpecificationFile should be a child class of the one called Specification that is like an interface because we should be able to manage another type of input, not only a local file (http request, for example) that's why I named this methos isNotValid, because due to its implementation, this should apply it own validation, for example, a remote file, should return a 200 Ok and not another type of response.

I hope that I have explained myself 😅

Reply:
Ok I understand :) I couldn't understand the context why `isNotValid` name. Maybe you should create the `Specification` interface and start using it, or even create the abstract Specification class? Something like:

```ts
interface Specification {
  isNotValid(): boolean;
}

// or

abstract class Specification {
  abstract isNotValid(): boolean;
}
```

and then implement it in `SpecificationFile`:

```ts
// you can also implements the abstract class
class SpecificationFile implements Specification {}
// or
class SpecificationFile extend Specification {}
```

Please consider it :)

Reply:
Yeap! I was waiting to the moment we really need it, but you are right, we can create it right now and prepare it for the future 😋